lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -1151,12 +1156,13 @@ function (ReflectionType $unionedType) use ($method, $parameter) {
             );
         }
 
-        if (! $type->isBuiltin()) {
+        if (! $type->isBuiltin() && $name !== 'static') {
             $name = '\\' . $name;
         }
 
         if (
-            $type->allowsNull()
+            ! $isUnionPart
+            && $type->allowsNull()
             && ($parameter === null || ! $parameter->isDefaultValueAvailable() || $parameter->getDefaultValue() !== null)
             && $name !== 'mixed'
```

Comment:
Thanks for working on this just found the nullable issue myself. 

Would it perhaps be better to avoid creating a new parameter and just check for name null like it is done for mixed?

```php
if (
    $type->allowsNull()
    && !in_array($name, ['mixed', 'null'], true)
    && ($parameter === null || ! $parameter->isDefaultValueAvailable() || $parameter->getDefaultValue() !== null)
) 
```

Reply:
Thanks for the suggestion, it's more logical :) I've fixed this with a new commit